batchflow/models/tf/layers/conv_block.py

```diff
@@ -159,33 +143,296 @@ def conv_block(inputs, layout='', filters=0, kernel_size=3, name=None,
+            return inputs
+
+        context = None
+        if self.name is not None:
+            context = tf.variable_scope(self.name, reuse=self.kwargs.get('reuse'))
+            context.__enter__()
+
+        layout_dict = {}
+        for layer in layout:
+            if self.C_GROUPS[layer] not in layout_dict:
+                layout_dict[self.C_GROUPS[layer]] = [-1, 0]
+            layout_dict[self.C_GROUPS[layer]][1] += 1
+
+
+        residuals = []
+        tensor = inputs
+        for i, layer in enumerate(layout):
+
+            layout_dict[self.C_GROUPS[layer]][0] += 1
+            layer_name = self.C_LAYERS[layer]
+            layer_fn = self.FUNC_LAYERS[layer_name]
+
+            args = {}
+            call_args = {}
+
+            if layer == 'a':
+                args = dict(activation=self.activation)
+                layer_fn = unpack_args(args, *layout_dict[self.C_GROUPS[layer]])['activation']
+                if layer_fn is not None:
+                    tensor = layer_fn(tensor)
+            elif layer == 'R':
+                residuals += [tensor]
+            elif layer == 'A':
+                args = dict(factor=self.kwargs.get('factor'), data_format=self.data_format)
+                args = unpack_args(args, *layout_dict[self.C_GROUPS[layer]])
+                t = self.FUNC_LAYERS['resize_bilinear_additive'](tensor, **args, name='rba-%d' % i)
+                residuals += [t]
+            elif layer == '+':
+                tensor = tensor + residuals[-1]
+                residuals = residuals[:-1]
+            elif layer == '.':
+                axis = -1 if self.data_format == 'channels_last' else 1
+                tensor = tf.concat([tensor, residuals[-1]], axis=axis, name='concat-%d' % i)
+                residuals = residuals[:-1]
+            else:
+                layer_args = self.kwargs.get(layer_name, {})
+                skip_layer = layer_args is False or isinstance(layer_args, dict) and layer_args.get('disable', False)
+
+                if skip_layer:
+                    pass
+                elif layer == 'f':
+                    if tensor.shape.ndims > 2:
+                        tensor = Flatten()(tensor)
+                    units = self.kwargs.get('units')
+                    if units is None:
+                        raise ValueError('units cannot be None if layout includes dense layers')
+                    args = dict(units=units)
+
+                elif layer == 'c':
+                    args = dict(filters=self.filters, kernel_size=self.kernel_size, strides=self.strides,
+                                padding=self.padding, data_format=self.data_format,
+                                dilation_rate=self.dilation_rate)
+                    if self.filters is None or self.filters == 0:
+                        raise ValueError('filters cannot be None or 0 if layout includes convolutional layers')
```

Comment:
What about making these 2 lines a function `check_filters_for_convolution`?

Reply:
Moved it to `conv`-related classes. I can move it to `ConvLayer` class and inherit all of the `conv` related classes from it, but that would not be looking all that nice. Will try though